src/zcbor_decode.c

```diff
@@ -964,48 +964,32 @@ bool zcbor_any_skip(zcbor_state_t *state, void *result)
 	INITIAL_CHECKS();
 	zcbor_major_type_t major_type = MAJOR_TYPE(*state->payload);
 	uint8_t additional = ADDITIONAL(*state->payload);
-	size_t value;
-	uint_fast32_t num_decode;
-	uint_fast32_t temp_elem_count;
-	uint_fast32_t elem_count_bak = state->elem_count;
-	uint8_t const *payload_bak = state->payload;
-	uint64_t tag_dummy;
-
-	payload_bak = state->payload;
-
-	if (!zcbor_multi_decode(0, ZCBOR_LARGE_ELEM_COUNT, &num_decode,
-			(zcbor_decoder_t *)zcbor_tag_decode, state,
-			(void *)&tag_dummy, 0)) {
-		state->elem_count = elem_count_bak;
-		state->payload = payload_bak;
-		ZCBOR_FAIL();
-	}
+	uint64_t value;
+	uint32_t tag_dummy;
+	zcbor_state_t local_copy;
 
-	if ((major_type == ZCBOR_MAJOR_TYPE_MAP) || (major_type == ZCBOR_MAJOR_TYPE_LIST)) {
-		if (additional == ZCBOR_VALUE_IS_INDEFINITE_LENGTH) {
-			ZCBOR_ERR_IF(state->elem_count == 0, ZCBOR_ERR_LOW_ELEM_COUNT);
-			state->payload++;
-			state->elem_count--;
-			temp_elem_count = state->elem_count;
-			payload_bak = state->payload;
-			state->elem_count = ZCBOR_LARGE_ELEM_COUNT;
-			if (!zcbor_multi_decode(0, ZCBOR_LARGE_ELEM_COUNT, &num_decode,
-					(zcbor_decoder_t *)zcbor_any_skip, state,
-					NULL, 0)
-					|| (state->payload >= state->payload_end)
-					|| !(*(state->payload++) == 0xFF)) {
-				state->elem_count = elem_count_bak;
-				state->payload = payload_bak;
-				ZCBOR_FAIL();
-			}
-			state->elem_count = temp_elem_count;
-			return true;
+	memcpy(&local_copy, state, sizeof(zcbor_state_t));
+
+	while (major_type == ZCBOR_MAJOR_TYPE_TAG) {
+		if (!zcbor_tag_decode(&local_copy, &tag_dummy)) {
+			ZCBOR_FAIL();
 		}
+		ZCBOR_ERR_IF(local_copy.payload >= local_copy.payload_end, ZCBOR_ERR_NO_PAYLOAD);
+		major_type = MAJOR_TYPE(*local_copy.payload);
+		additional = ADDITIONAL(*local_copy.payload);
 	}
 
-	if (!value_extract(state, &value, sizeof(value))) {
-		/* Can happen because of elem_count (or payload_end) */
-		ZCBOR_FAIL();
+	bool indefinite_length_array = (additional == ZCBOR_VALUE_IS_INDEFINITE_LENGTH);
+
+	if (((major_type == ZCBOR_MAJOR_TYPE_MAP) || (major_type == ZCBOR_MAJOR_TYPE_LIST))
+			&& indefinite_length_array) {
+		local_copy.payload++;
+		value = ZCBOR_LARGE_ELEM_COUNT;
```

Comment:
`major_type` is passed into `switch case` block in the further part of the function and `value` is overwritten there in case of `..TYPE_MAP`. What about handling everything there? That would allow to decrease nesting of the `else if` statement as well.